api-controllers/ServiceBrokerApiController.js

```diff
@@ -294,6 +294,13 @@ class ServiceBrokerApiController extends FabrikBaseController {
       logger.debug('returning ..', body);
       return Promise.try(() => {
         if (_.get(operation, 'type') === 'delete' && body.state === CONST.OPERATION.SUCCEEDED && resourceGroup === CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR) {
+          // consistency check that a successfully deleted resource has no resources attached
+          if (!_.isEmpty(result.resources)) {
```

Comment:
@steinri 
This change will put the instance in inconsistent state. Ideally when the state is set to succeeded, finalizer of interoperator is also removed from sfserviceinstance CR [(ref.)](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/interoperator/pkg/controller/sfserviceinstance/sfserviceinstance_controller.go#L371-L376).

When next time delete is tried on the instance, broker will set the state as delete but now interoperator does not act on this sfserviceinstance as interoperator finalizer was already removed. It means that state of sfserviceinstance will always be delete in progress.

Ideally what you are reporting should not have happened. I'd suggest that we figure out the root cause in case it pops up again and fix it there.

Reply:
I agree that the root cause should be fixed and that this might not be the correct implementation for the desired behavior.
However I think that the service fabrik should be a little more robust to such cases. In general it might be a better solution to store the last operation in the sfserviceinstance CR and compare it to the operation from the polling request instead of trusting the Cloud Controller to send the correct last operation.